src/datalad_installer.py

```diff
@@ -2076,20 +2076,26 @@ def install_deb(
     install_dir: Optional[Path] = None,
     extra_args: Optional[List[str]] = None,
 ) -> Path:
-    cmd: List[Union[str, os.PathLike]] = ["dpkg"]
-    if extra_args is not None:
-        cmd.extend(extra_args)
     if install_dir is None:
+        cmd: List[Union[str, os.PathLike]] = ["dpkg"]
+        if extra_args is not None:
+            cmd.extend(extra_args)
         cmd.append("-i")
         cmd.append(debpath)
         manager.sudo(*cmd)
         return Path("/usr/bin")
     else:
+        if extra_args:
+            log.warning("Not using dpkg; ignoring --extra-args")
+        assert os.path.isabs(debpath)
         install_dir.mkdir(parents=True, exist_ok=True)
-        cmd.append("-x")
-        cmd.append(debpath)
-        cmd.append(install_dir)
-        runcmd(*cmd)
+        install_dir = install_dir.resolve()
+        with tempfile.TemporaryDirectory() as tmpdir:
+            oldpwd = os.getcwd()
+            os.chdir(tmpdir)
```

Comment:
note: although unlikely to matter here `chdir` within a process is a bit "atrocious" , e.g. IIRC `getcwd` would return a resolved directory name if you are in a dir which is symlinked somewhere.  I thought to suggest to use `--output` option for `ar` I find available on Debian  2.35.2, but it seems to be not available on centos with 2.27-27.base.el7.  So let's keep `chdir`ing I guess